internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -301,7 +302,10 @@ func (c *enrollCmd) writeDelayEnroll(streams *cli.IOStreams) error {
 func (c *enrollCmd) fleetServerBootstrap(ctx context.Context, persistentConfig map[string]interface{}) (string, error) {
 	c.log.Debug("verifying communication with running Elastic Agent daemon")
 	agentRunning := true
-	_, err := getDaemonState(ctx)
+	if c.options.FleetServer.InternalPort == 0 {
+		c.options.FleetServer.InternalPort = defaultFleetServerInternalPort
+	}
```

Comment:
The InternalURL is set when the agent is creating the fleet tls settings ([line 321](https://github.com/elastic/elastic-agent/pull/1867/files#diff-7efa04d4079650519d21a6e8ba217a7874130cecf8adc92dc2a78d4cfd2aee09R321)), however it expects a non-zero internal port value in order to do so.

Reply:
this is fine. we have a check like this inside `createFleetServerBootstrapConfig` i'm ok with moving this up.